### PR TITLE
drivers: display: st7735r: added configuration for color inverting

### PR DESCRIPTION
**Pull request by soburi**
Add 'invert-color' configuration to st7735r dts-node.

Issue INVON(21h) command on initializing if invert-color was enabled.
As a result of this command, the display color is inverted.
Otherwise, INVOFF(20h) will be issued.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

#44796 is depends to this patch.

### CODE REVIEW

File: dts/bindings/display/sitronix,st7735r.yaml [hunk 1]
```diff
@@ -115,3 +115,7 @@ properties:
       type: int
       default: 0x07
       description: Display Inversion Control
+
+    invert-color:
+      type: boolean
+      description: Display color inversion
```
Comment (jfischer-no) [inline dts/bindings/display/sitronix,st7735r.yaml:121]: Since you are touching it, can you please fix description from `invctr` above to "Display Inversion mode"? If so please do it in separate commit.

The name and description of property you would like to add should probably be:

```
    inversion-on:
      type: boolean
      description: Enable Display Inversion
```

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: dts/bindings/display/sitronix,st7735r.yaml (hunk 1 above)
Comment (soburi) [reply dts/bindings/display/sitronix,st7735r.yaml]: Splitting commit and Fix it.
Changing variable name as well.

Comment (jfischer-no) [reply dts/bindings/display/sitronix,st7735r.yaml]: >Since you are touching it, can you please fix description from `invctr` above to "Display Inversion mode"? If so please do it in separate commit.

You do not need to split your commit. I asked if you could fix the description of existing `invctr` property because it is a bit confusing now, if so, it should be done in separate commit.

**Review by mbolivar-nordic — COMMENTED** [review]
LGTM, @jfischer-no please take another look

**Review by mbolivar-nordic — DISMISSED** [review]

File: dts/bindings/display/sitronix,st7735r.yaml (hunk 1 above)
Comment (soburi) [reply dts/bindings/display/sitronix,st7735r.yaml]: I got it.
I updated description of 'invctr' with separeted commit with  adding short summary of datasheet's description.

File: dts/bindings/display/sitronix,st7735r.yaml [hunk 2]
```diff
@@ -117,3 +117,9 @@ properties:
       description: Display Inversion Control
 
         Set dot inversion or line inversion for each normal/idle/partial mode.
+
+    inversion-on:
+      type: boolean
+      description: Enable Display Inversion
```
Comment (jfischer-no) [inline dts/bindings/display/sitronix,st7735r.yaml:123]: Not sure about style here, we have different in the tree, but @mbolivar-nordic will complain if necessary.
```
      description: |

        Enable Display Inversion
        Make a drawing with the inverted color of the frame memory.
```

**Review by jfischer-no — DISMISSED** [review]

File: dts/bindings/display/sitronix,st7735r.yaml (hunk 2 above)
Comment (mbolivar-nordic) [reply dts/bindings/display/sitronix,st7735r.yaml:123]: @jfischer-no is right; this needs to be done in a different way. Like this, please:

```
description: |
    Enable Display Inversion
    Make a drawing with the inverted color of the frame memory.
```

The same change should be made for the `invctr` property also.

Please see https://docs.zephyrproject.org/latest/build/dts/bindings.html#general-rules for an explanation about why this is needed.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]

File: dts/bindings/display/sitronix,st7735r.yaml (hunk 2 above)
Comment (soburi) [reply dts/bindings/display/sitronix,st7735r.yaml]: @jfischer-no , @mbolivar-nordic 
Thanks for reviewing.
I Fixed it.

**Review by jfischer-no — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]
